code/game/objects/items/weapons/RSF.dm

```diff
@@ -232,3 +232,76 @@ RSF
 				to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
 				desc = "A RSF. It currently holds [matter]/30 fabrication-units."
 		return
+
+/obj/item/weapon/cookiesynth
+	name = "Cookie Synthesizer"
+	desc = "A self-recharging device used to rapidly deploy cookies."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcd"
+	var/matter = 10
+	var/toxin = 0
+	var/cooldown = 0
+	var/cooldowndelay = 10
+	var/emagged = 0
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/weapon/cookiesynth/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>It currently holds [matter]/10 cookie-units.</span>")
+
+/obj/item/weapon/cookiesynth/attackby()
+	return
+
+/obj/item/weapon/cookiesynth/emag_act(mob/user)
+	emagged = !emagged
+	if(emagged)
+		to_chat(user, "<span class='warning'>You short out the [src]'s reagent safety checker!</span>")
+	else
+		to_chat(user, "<span class='warning'>You reset the [src]'s reagent safety checker!</span>")
+		toxin = 0
+
+/obj/item/weapon/cookiesynth/attack_self(mob/user)
+	var/mob/living/silicon/robot/P = null
+	if(isrobot(user))
+		P = user
+	if(emagged && !toxin)
+		toxin = 1
+		to_chat(user, "Cookie Synthesizer Hacked")
+	else if(P.emagged && !toxin)
+		toxin = 1
+		to_chat(user, "Cookie Synthesizer Hacked")
+	else
+		toxin = 0
+		to_chat(user, "Cookie Synthesizer Reset")
+
+/obj/item/weapon/cookiesynth/process()
+	if(matter < 10)
+		matter++
+
+/obj/item/weapon/cookiesynth/afterattack(atom/A, mob/user, proximity)
+	if(cooldown > world.time)
+		return
+	if(!proximity)
+		return
+	if (!(istype(A, /obj/structure/table) || isfloorturf(A)))

```

Comment:
This is probably going to be confusing if someone just wants to give someone a cookie

Reply:
currently this functions just like the Cash, Dice, and drinking glass dispensers; they all function like this.